### PR TITLE
move verbose sections in tutorials to explanation

### LINKED ISSUE #380: Move verbose tutorial sections to explanatory
Some blocks in our tutorials nearly take up an entire page, and I think it detracts from the code itself. Let's move some of these longer sections into sections in the explanatory material.

### PR DESCRIPTION
**Pull request by caleb-johnson**
Fixes #380

### CODE REVIEW

File: docs/circuit_cutting/explanation/index.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.. _circuit cutting explanation:
+.. _Circuit cutting explanation:
```
[the remark this answers is not in the document]
Comment (caleb-johnson) [reply docs/circuit_cutting/explanation/index.rst:1]: I think I removed it during debugging and put it back incorrectly. Surprised it didnt break something

File: docs/circuit_cutting/tutorials/01_gate_cutting_to_reduce_circuit_width.ipynb
Comment (garrison) [inline docs/circuit_cutting/tutorials/01_gate_cutting_to_reduce_circuit_width.ipynb:1]: > one weight is associated with each sample

I found this confusing the first time I read it, but on second read, it's not _so_ bad.  Perhaps could be changed to "one combined weight is given for each unique sample".

**Review by garrison — COMMENTED** [review]
Aside from my one comment, this looks great to me.  Thanks for making these adjustments.

File: docs/circuit_cutting/tutorials/02_gate_cutting_to_reduce_circuit_depth.ipynb
Comment (garrison) [inline docs/circuit_cutting/tutorials/02_gate_cutting_to_reduce_circuit_depth.ipynb:1]: Could you please make the same change (https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/pull/381#discussion_r1311699312) in the other two files as well?

**Review by garrison — APPROVED** [review]
Thanks, looks great!

**Pull request merged**